src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets

```diff
@@ -82,23 +89,8 @@ properties that determine build ordering.
 
   <PropertyGroup Condition=" '$(AndroidApplication)' != 'True' ">
     <BuildDependsOn>
```

Comment:
It looks like previously we had a different value for `$(BuildDependsOn)` for class libraries vs apps. Is this now sharing the same set of targets for both?

I think there were a few differences in there that were important.

Reply:
There were differences I can't remember what though.

Reply:
We do still have two groups for libraries vs apps. Those groups are now much smaller, as the bulk of the targets declared in both groups were duplicated. The new `AndroidBuildDependsOn` group contains the targets used by both project types.

The PR description has info about the project type specific targets that were modified, removed, or now run for both project types as part of these changes. The only _potential_ behavior changes here would be with respect to `_CheckProjectItems `, and `_RemoveLegacyDesigner` as described in the PR.

The reason behind these changes is that I was looking to add a new target for all project types, and I wasn't sure where the best place to put it would be. It also seemed strange to have copy that target to two lists.

I'm happy to abandon this approach and go with something else if there is abetter way to add a target to run before _CategorizeAndroidLibraries.

Reply:
For reference, here's diff of the two targets groups as they exist on main currently:

```diff
--- App-BuildDependsOn.targets
+++ Lib-BuildDependsOn.targets
@@ -1,21 +1,22 @@
+<PropertyGroup Condition=" '$(AndroidApplication)' != 'True' ">
     <BuildDependsOn>
       _ValidateLinkMode;
-      _CheckNonIdealConfigurations;
+      _CheckNonIdealBindingConfigurations;
       _SetupMSBuildAllProjects;
       _SetupDesignTimeBuildForBuild;
       _CategorizeAndroidLibraries;
       _CreatePropertiesCache;
       _CleanIntermediateIfNeeded;
-      _CheckProjectItems;
+      _AddAndroidDefines;
       _CheckForContent;
       _CheckForObsoleteFrameworkAssemblies;
+      _CalculateAssetsWithAssetPackMetaData;
+      _RemoveLegacyDesigner;
       _ValidateAndroidPackageProperties;
       AddLibraryJarsToBind;
       _CollectJavaSourceForBinding;
       _CompileBindingJava;
       $(BuildDependsOn);
-      _CompileDex;
-      $(_AfterCompileDex);
+      _CreateAar;
     </BuildDependsOn>
 </PropertyGroup>
```

Reply:
We don't want `_CreateAar` running for apps, and we wouldn't want `_CompileDex` running for class libraries.

Adding `$(AndroidBuildDependsOn)` is a good idea, but I think we should try to keep the appropriate targets for both project types.

Maybe we move the condition to the targets?

Reply:
_CreateAar and _CompileDex are still separated. The GitHub diff doesn't show this well... Looking at the file itself through here are the new splits:

Shared AndroidBuildDependsOn:(https://github.com/dotnet/android/blob/a535c4f56ac2ec7c60ac9aa89b0f35aa4085ef58/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets#L14-L31)

App BuildDependsOn:
https://github.com/dotnet/android/blob/a535c4f56ac2ec7c60ac9aa89b0f35aa4085ef58/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets#L34-L39

Library BuildDependsOn:
https://github.com/dotnet/android/blob/a535c4f56ac2ec7c60ac9aa89b0f35aa4085ef58/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets#L91-L94